Report Diff Summary Improvements

Vampire opened the pull request:
1. add a title to the report diff summary page
2. add tested projects count to report diff summary
3. add checkstyle favicon to report diff summary
4. clear report diff summary if it exists already
5. sort report diff summary, projects with differences first, case-insensitive alphabetical within the two sections
6. add total found differences to report diff summary

![grafik](https://user-images.githubusercontent.com/325196/54827024-59248a00-4cb1-11e9-9c78-e2c5a52fec13.png)

rnveach commented on checkstyle-tester/diff.groovy:
```diff
@@ -300,6 +304,11 @@ def printReportInfoSection(summaryIndexHtml, checkstyleBaseReportInfo, checkstyl
+    summaryIndexHtml << ('<br />')
+    summaryIndexHtml << "Tested projects: ${projectsStatistic.size()}"
+    summaryIndexHtml << ('<br />')
+    summaryIndexHtml << "Total found differences: ${projectsStatistic.values().sum()}"
```
`Total found differences` => `Total differences found`

romani commented:
Please provide pictures on how it was before and it becomes, to let more easily validate your PR.

Change is big enough, we need to reference PR number in commit messages.

Vampire commented:
> Please provide pictures on how it was before and it becomes, to let more easily validate your PR.

added to original post

> Change is big enough, we need to reference PR number in commit messages.

changed
